object_visualizer/src/object_visualizer_node.cpp

```diff
@@ -127,13 +135,33 @@ namespace object_visualizer
         marker.color.a = 1.0;
       }
 
+      int marker_size_x = obj.size.x;
+      int marker_size_y = obj.size.y;
+      int marker_size_z = obj.size.z;
+
+      marker.pose = obj.pose.pose;
+
+      // overwrite size in case any previous stack doesn't provide size
+      // such as carma_cooperative_perception at the moment
+      if (obj.size.x < 0.01 || obj.size.y < 0.01 || obj.size.z < 0.01)
+      {
+        marker_size_x = 1.0;
+        marker_size_y = 1.0;
+        marker_size_z = 1.0;
+        marker.pose.position.z = 1.0; // lifts the marker above ground so that it doesn't clip
+      }
+
       marker.id = id;
-      marker.type = visualization_msgs::msg::Marker::CUBE;
+      if (static_cast<int>(config_.marker_shape) < 1 || static_cast<int>(config_.marker_shape) > 3)
+      {
+        throw std::invalid_argument("Marker shape is not valid: " + std::to_string(static_cast<int>(config_.marker_shape)) + ". Please choose from 1: CUBE, 2: SPHERE, 3: CYLINDER");
+      }
+      marker.type = config_.marker_shape;
       marker.action = visualization_msgs::msg::Marker::ADD;
-      marker.pose = obj.pose.pose;
-      marker.scale.x = obj.size.x * 2.0; // Size in carma is half the length/width/height 
-      marker.scale.y = obj.size.y * 2.0;
-      marker.scale.z = obj.size.z * 2.0;
+
+      marker.scale.x = marker_size_x * 2; // Size in carma is half the length/width/height
+      marker.scale.y = marker_size_y * 2;
+      marker.scale.z = marker_size_z * 2;
```

Comment:
The marker scale/size can be simplified, assuming we want a minimum marker size of 1.0:
```cpp
marker.scale.x = std::max(1.0, obj.size.x) * 2;
marker.scale.y = std::max(1.0, obj.size.y) * 2;
marker.scale.z = std::max(1.0, obj.size.z) * 2;
```

Reply:
incorporated